crates/bevy_reflect/src/serde/ser.rs

```diff
@@ -317,8 +317,16 @@ impl<'a> Serialize for EnumSerializer<'a> {
 
         match variant_type {
             VariantType::Unit => {
-                if self.enum_value.reflect_module_path() == Some("core::option")
-                    && self.enum_value.reflect_type_ident() == Some("Option")
+                if self
+                    .enum_value
+                    .get_represented_type_info()
```

Comment:
Actually I just realized that we get the represented type info a few lines up. We should reuse that value to reduce the performance cost of calling `Reflect::get_represented_type_info`.

Reply:
This PR looks good to me other than this suggestion :) Let me know when that's done and I can merge it in.

Reply:
fixed